pom.xml

```diff
@@ -234,8 +234,57 @@
     <profiles>
         <profile>
             <id>native</id>
+            <dependencies>
+                <dependency>
+                    <!-- needs to be correct version fitting to junit-jupiter -->
+                    <groupId>org.junit.platform</groupId>
+                    <artifactId>junit-platform-launcher</artifactId>
+                    <version>1.9.2</version>
+                    <scope>test</scope>
+                </dependency>
+            </dependencies>
             <build>
                 <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>exec-maven-plugin</artifactId>
+                        <version>1.6.0</version>
+                        <executions>
+                            <execution>
+                                <id>run-script</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <executable>${project.basedir}/graalvm/generate-config.sh</executable>
+                                    <arguments>
+                                        <argument>-t</argument>
+                                        <argument>${project.build.directory}</argument>
+                                    </arguments>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>cleanup-files</id>
+                                <phase>test</phase>
```

Comment:
why do you run that in the test phase?

Reply:
If you want to run the test as a native-binary, you can execute `mvn -Pnative -Dagent=true test`. This will first run the tests normally with the tracing agent from graalvm. After the test run, the native binary is compiled with the generated configurations from the previous run. 
This happens during the test phase of the native profile. But because of the bug, where the generated configurations of the tracing agent are wrongly formatted, this command needs to be executed first to cleanup the configurations, before they can be used for compilation.